subscription/src/main/java/org/killbill/billing/subscription/engine/dao/DefaultSubscriptionDao.java

```diff
@@ -577,7 +578,9 @@ public List<SubscriptionBaseEvent> getPendingEventsForSubscription(final UUID su
         return transactionalSqlDao.execute(true, new EntitySqlDaoTransactionWrapper<List<SubscriptionBaseEvent>>() {
             @Override
             public List<SubscriptionBaseEvent> inTransaction(final EntitySqlDaoWrapperFactory entitySqlDaoWrapperFactory) throws Exception {
-                final List<SubscriptionEventModelDao> eventModels = entitySqlDaoWrapperFactory.become(SubscriptionEventSqlDao.class).getFutureActiveEventForSubscription(subscriptionId.toString(), now, context);
+                final SubscriptionEventSqlDao eventSqlDao = entitySqlDaoWrapperFactory.become(SubscriptionEventSqlDao.class);
+                final List<SubscriptionEventModelDao> eventModels = eventSqlDao.getFutureActiveEventForSubscription(subscriptionId.toString(), now, context);
```

Comment:
Nit: I suspect changing the DAO methods to return a `SortedSet` instead would return an ordered collection automagically. That way, no need to call manually `Collections.sort` each time.

Reply:
Great point. Fixed in https://github.com/killbill/killbill/pull/1673/commits/2e16492b0b219d98ec28f4139a8768d9ea43e0fc

Reply:
Also created [GH issue] to track my questions.